### PR TITLE
feat(frontend): display token as logo above qr-code

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

The designers want to display the token logo above the qr-codes.

# Changes

- Pass the token from every consumers to the QR code
- Remove `undefined` assignation of `addressToken` provided in #3024 given that the property now is always set

# Screenshots

Capture d’écran 2024-10-22 à 08 31 50
Capture d’écran 2024-10-22 à 08 31 34
Capture d’écran 2024-10-22 à 08 31 31
Capture d’écran 2024-10-22 à 08 31 28
Capture d’écran 2024-10-22 à 08 31 22
Capture d’écran 2024-10-22 à 08 31 19

### CODE REVIEW

**Review by AntonioVentilii-DFINITY — APPROVED** [review]
LGTM! Tks!

Three questions of the same nature, but non-blockers

File: src/frontend/src/lib/components/receive/ReceiveModal.svelte [hunk 1]
```diff
@@ -7,8 +7,10 @@
 	import { i18n } from '$lib/stores/i18n.store';
 	import { modalStore } from '$lib/stores/modal.store';
 	import type { OptionAddress, Address } from '$lib/types/address';
+	import type { Token } from '$lib/types/token';
 
 	export let address: OptionAddress<Address> = undefined;
+	export let addressToken: Token | undefined = undefined;
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/lib/components/receive/ReceiveModal.svelte:13]: I guess same answer as https://github.com/dfinity/oisy-wallet/pull/3023#discussion_r1811824886.

File: src/frontend/src/lib/components/receive/ReceiveAddressQRCode.svelte [hunk 2]
```diff
@@ -6,9 +6,11 @@
 	import Copy from '$lib/components/ui/Copy.svelte';
 	import { RECEIVE_TOKENS_MODAL_QR_CODE_OUTPUT } from '$lib/constants/test-ids.constants';
 	import { i18n } from '$lib/stores/i18n.store';
+	import type { Token } from '$lib/types/token';
 
 	export let address: undefined | string;
 	export let addressLabel: string | undefined = undefined;
+	export let addressToken: Token | undefined;
```
[the remark this answers is not in the document]
Comment (peterpeterparker) [reply src/frontend/src/lib/components/receive/ReceiveAddressQRCode.svelte:13]: ditto https://github.com/dfinity/oisy-wallet/pull/3023#discussion_r1811824886

File: src/frontend/src/lib/components/receive/ReceiveQRCode.svelte [hunk 3]
```diff
@@ -8,8 +8,7 @@
 	import { replacePlaceholders } from '$lib/utils/i18n.utils';
 
 	export let address: string;
-	// TODO: remove undefined assignation to make property mandatory in PR #3023
-	export let addressToken: Token | undefined = undefined;
+	export let addressToken: Token | undefined;
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/receive/ReceiveQRCode.svelte]: All good! thank you for the explanation!

**Pull request merged**